extensions-core/druid-kerberos/src/main/java/io/druid/kerberos/KerberosHttpClient.java

```diff
@@ -0,0 +1,141 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.kerberos;
+
+import com.google.common.base.Strings;
+import com.google.common.base.Throwables;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.metamx.http.client.AbstractHttpClient;
+import com.metamx.http.client.HttpClient;
+import com.metamx.http.client.Request;
+import com.metamx.http.client.response.HttpResponseHandler;
+import io.druid.java.util.common.ISE;
+import io.druid.java.util.common.logger.Logger;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.CommonConfigurationKeysPublic;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.ietf.jgss.GSSContext;
+import org.ietf.jgss.GSSException;
+import org.ietf.jgss.GSSManager;
+import org.ietf.jgss.GSSName;
+import org.ietf.jgss.Oid;
+import org.jboss.netty.handler.codec.http.HttpHeaders;
+import org.joda.time.Duration;
+
+import java.io.IOException;
+import java.security.PrivilegedExceptionAction;
+
+public class KerberosHttpClient extends AbstractHttpClient
+{
+  private static final Logger log = new Logger(KerberosHttpClient.class);
+
+  private final HttpClient delegate;
+  private final DruidKerberosConfig config;
+  private static final Base64 base64codec = new Base64(0);
+
+
+  public KerberosHttpClient(HttpClient delegate, DruidKerberosConfig config)
+  {
+    this.delegate = delegate;
+    this.config = config;
+  }
+
+  @Override
+  public <Intermediate, Final> ListenableFuture<Final> go(
+    Request request, HttpResponseHandler<Intermediate, Final> httpResponseHandler, Duration duration
+  )
+  {
+    try {
+      final String host = request.getUrl().getHost();
+      authenticateIfRequired();
+      UserGroupInformation currentUser = UserGroupInformation.getCurrentUser();
+      String challenge = currentUser.doAs(new PrivilegedExceptionAction<String>()
+      {
+        @Override
+        public String run() throws Exception
+        {
+          return kerberosChallenge(host);
+        }
+      });
+      request.setHeader(HttpHeaders.Names.AUTHORIZATION, "Negotiate " + challenge);
+    }
+    catch (Throwable e) {
+      Throwables.propagate(e);
+    }
+    return delegate.go(request, httpResponseHandler, duration);
+  }
+
+
+  public void authenticateIfRequired()
+    throws IOException
+  {
+    String principal = config.getPrincipal();
+    String keytab = config.getKeytab();
+    if (!Strings.isNullOrEmpty(principal) && !Strings.isNullOrEmpty(keytab)) {
+      Configuration conf = new Configuration();
+      conf.set(CommonConfigurationKeysPublic.HADOOP_SECURITY_AUTHENTICATION, "kerberos");
+      UserGroupInformation.setConfiguration(conf);
+      if (UserGroupInformation.isSecurityEnabled()) {
```

Comment:
@nishantmonu51t this seems to me it is using hadoop conf files to check of kerberos is enabled. would this still work in case we are not using hadoop config with druid ?